backend/matching-service/app/src/matching/matching.interfaces.ts

```diff
@@ -0,0 +1,40 @@
+import { Socket } from "socket.io";
+
+const DIFFICULTY = {
+  EASY: "Easy",
+  MEDIUM: "Medium",
+  HARD: "Hard",
+} as const;
+
+type DifficultyType = (typeof DIFFICULTY)[keyof typeof DIFFICULTY];
+
+const TOPIC_TAG = {
```

Comment:
There are a lot more topics than this at the moment available in the Questions DB: Maybe we should standardise what topics we want to have available.

Just for reference:
``` 
"topics": [
    "Array",
    "Backtracking",
    "Binary Indexed Tree",
    "Binary Search",
    "Binary Search Tree",
    "Binary Tree",
    "Bit Manipulation",
    "Bitmask",
    "Brainteaser",
    "Breadth-First Search",
    "Bucket Sort",
    "Combinatorics",
    "Counting",
    "Counting Sort",
    "Data Stream",
    "Database",
    "Depth-First Search",
    "Design",
    "Divide and Conquer",
    "Doubly-Linked List",
    "Dynamic Programming",
    "Enumeration",
    "Eulerian Circuit",
    "Game Theory",
    "Geometry",
    "Graph",
    "Greedy",
    "Hash Function",
    "Hash Table",
    "Heap (Priority Queue)",
    "Interactive",
    "Iterator",
    "Line Sweep",
    "Linked List",
    "Math",
    "Matrix",
    "Memoization",
    "Merge Sort",
    "Monotonic Queue",
    "Monotonic Stack",
    "Number Theory",
    "Ordered Set",
    "Prefix Sum",
    "Probability and Statistics",
    "Queue",
    "Quickselect",
    "Radix Sort",
    "Randomized",
    "Recursion",
    "Reservoir Sampling",
    "Rolling Hash",
    "Segment Tree",
    "Shell",
    "Shortest Path",
    "Simulation",
    "Sliding Window",
    "Sorting",
    "Stack",
    "String",
    "String Matching",
    "Topological Sort",
    "Tree",
    "Trie",
    "Two Pointers",
    "Union Find"
  ]
```

Reply:
yea i think we should make a decision whether to
1. hardcode the list of topics
2. dynamically retrieve via a question service API

Reply:
removed the list of topics from the backend to just be an array of strings. backend doesn't need to know what type of questions there are. 6d9abf736fe9ae8b905c31128d1f06024114dc9d